src/combinatorics/young_diagram.lean

```diff
@@ -123,4 +128,128 @@ end distrib_lattice
 /-- Cardinality of a Young diagram -/
 @[reducible] protected def card (μ : young_diagram) : ℕ := μ.cells.card
 
+section transpose
+/-- The `transpose` of a Young diagram is obtained by swapping i's with j's. --/
+
+def transpose (μ : young_diagram) : young_diagram :=
+{ cells :=  (equiv.prod_comm _ _).finset_congr μ.cells,
+  is_lower_set := λ _ _ h, by {
+    simp only [finset.mem_coe, equiv.finset_congr_apply, finset.mem_map_equiv],
+    intro hcell, apply μ.is_lower_set _ hcell, simp [h], } }
```

Comment:
This linter error is that curly braces aren't allowed at the end of a line. This would be allowed:
```suggestion
  is_lower_set := λ _ _ h,
    by { simp only [finset.mem_coe, equiv.finset_congr_apply, finset.mem_map_equiv],
         intro hcell, apply μ.is_lower_set _ hcell, simp [h], } }
```
You might consider using `begin`/`end` though for readability:
```suggestion
  is_lower_set := λ _ _ h, begin
    simp only [finset.mem_coe, equiv.finset_congr_apply, finset.mem_map_equiv],
    intro hcell,
    apply μ.is_lower_set _ hcell,
    simp [h],
  end }
```